libyul/backends/evm/EVMBuiltins.cpp

```diff
@@ -0,0 +1,418 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <libyul/backends/evm/EVMBuiltins.h>
+
+#include <libyul/AST.h>
+#include <libyul/Object.h>
+#include <libyul/Utilities.h>
+
+#include <libevmasm/AssemblyItem.h>
+
+#include <libsolutil/StringUtils.h>
+
+#include <range/v3/algorithm/all_of.hpp>
+
+using namespace solidity;
+using namespace solidity::yul;
+
+namespace
+{
+
+BuiltinFunctionForEVM createFunction(
+		std::string const& _name,
+		size_t _params,
+		size_t _returns,
+		SideEffects _sideEffects,
+		ControlFlowSideEffects _controlFlowSideEffects,
+		std::vector<std::optional<LiteralKind>> _literalArguments,
+		std::function<void(FunctionCall const&, AbstractAssembly&, BuiltinContext&)> _generateCode
+	)
```

Comment:
Too much indent:
```suggestion
	std::function<void(FunctionCall const&, AbstractAssembly&, BuiltinContext&)> _generateCode
)
```

Same for `createFunction()` lambda in `createVerbatimFunction()`.

Reply:
And if we're at whitespace, please leave an empty line before `private` and `public` in the class definitions.